contracts/contracts/Subsidy.sol

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+
+import {IMACI} from "./IMACI.sol";
+import {MessageProcessor} from "./MessageProcessor.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {Poll} from "./Poll.sol";
+import {SnarkCommon} from "./crypto/SnarkCommon.sol";
+import {CommonUtilities} from "./Utility.sol";
+import {Verifier} from "./crypto/Verifier.sol";
+import {VkRegistry} from "./VkRegistry.sol";
+
+contract Subsidy is
+    Ownable,
+    CommonUtilities,
+    SnarkCommon
+{
+
+    uint256 public rbi; // row batch index
+    uint256 public cbi; // column batch index
+    // The final commitment to the state and ballot roots
+    uint256 public sbCommitment;
+    uint256 public subsidyCommitment;
+
+    // Error codes
+    string constant ERROR_VOTING_PERIOD_NOT_PASSED = "SubsidyE01";
+    string constant ERROR_PROCESSING_NOT_COMPLETE = "SubsidyE02";
+    string constant ERROR_INVALID_SUBSIDY_PROOF = "SubsidyE03";
+    string constant ERROR_ALL_SUBSIDY_CALCULATED = "SubsidyE04";
+    string constant ERROR_VK_NOT_SET = "SubsidyE05";
+
+    Verifier public verifier;
+
+    constructor(Verifier _verifier) {
+        verifier = _verifier;
+    }
+
+    modifier votingPeriodOver(Poll _poll) {
```

Comment:
could move this to a internal/private function. I see the same modifier is in Tally.sol so we could move it to an utilities contract (CommonUtilities) as internal and inherit from it and use it that way

Reply:
can modifier be inherited as well?

Reply:
you can yes, but we can also make it an internal function and inherit that way.